app/ide-desktop/client/tests/gettingStarted2.spec.ts

```diff
@@ -0,0 +1,221 @@
+      .click()
+
+    // Visualize and assert the result
+    await visualizeData(page)
+    await expect(page.getByText('1026')).toBeVisible()
+
+    // Adding sort component
+    await createComponentText(page, 'aggregate')
+    await page.locator('.ComponentEntry', { hasText: 'sort' }).click()
+
+    // Click the plus
+    await page
+      .locator('div.WidgetTopLevelArgument', { hasText: 'columns' })
+      .nth(1)
+      .getByRole('list')
+      .filter({ hasText: /^$/ })
+      .getByLabel('Add a new item')
+      .click()
+
+    // Choosing parameters
+    await page
+      .locator('div')
+      .filter({ hasText: /^‘product_name’$/ })
+      .nth(4)
+      .click()
+    await page.getByRole('button', { name: 'Count', exact: true }).click()
+
+    await page.getByText('direction', { exact: true }).click()
+    await page.getByRole('button', { name: '..Descending', exact: true }).click()
+
+    await visualizeData(page)
+  })
+
+  // ---------------- Objective 3 ----------------
+  await test.step('Objective 3: Create table of currencies by product names', async () => {
+    // Scroll into view
+    await page.mouse.wheel(0, -200)
+
+    // Creating cross_tab component
+    const readComponent = page.getByText('read', { exact: true }).nth(2)
+    await readComponent.click({ button: 'right' })
+
+    await page.keyboard.press('Enter')
+    await page.locator('.ComponentEntry', { hasText: 'cross_tab' }).click()
+
+    // Choosing the right parameters
+    const crossGroup = await page.getByText('group_by', { exact: true }).nth(1)
+
+    // Ensuring 'plus' is visible, to avoid clicking too early
+    await expect(page.locator('div').getByLabel('Add a new item').last()).toBeVisible()
+    await crossGroup.click()
+    await page.getByRole('button', { name: 'product_name', exact: true }).click()
+
+    await page.locator('.WidgetSelection.clickable').filter({ hasText: 'names' }).click()
+    const curRCode = page.getByRole('button', { name: 'currency_code' }).first()
+    await expect(curRCode).toBeVisible()
+    await curRCode.click()
+
+    await page.getByText('values', { exact: true }).click()
+    await page.getByRole('button', { name: '..Count_Distinct', exact: true }).click()
+
+    // Click the plus and select argument
+    const plus = page
+      .locator('div.WidgetTopLevelArgument', { hasText: 'columns' })
+      .getByRole('list')
+      .filter({ hasText: /^$/ })
+      .getByLabel('Add a new item')
+    await expect(plus).toBeVisible()
+    await plus.click()
+    await page.getByRole('button', { name: 'account_id', exact: true }).click()
+
+    await visualizeData(page)
+
+    // Move cross_tab a bit down for more clearance
+    const crossTab = page.getByText('cross_tab')
+
+    await crossTab.hover()
+    await page.mouse.down()
+    const box = await crossTab.boundingBox()
+    if (box) {
+      await page.mouse.move(box.x + box.width / 2, box.y + box.height / 2 + 200)
+    }
+    await page.mouse.up()
```

Comment:
There is an easier way to do dragging. Check out `moveNode` in `collapsingAndEntering.spec.ts`:
```ts
function moveNode(page: Page, binding: string, relativePos: RelativePos) {
    return grabNode(page, binding).dragTo(grabNode(page, relativePos.relativeTo ?? binding), {
      force: true,
      targetPosition: relativePos,
    })
  }
```

`grabNode` can be replaced with `graphNodeIcon`, it’s an oversight on my side.